app/models/course_template.rb

```diff
@@ -26,6 +27,9 @@ class CourseTemplate < ActiveRecord::Base
   scope :not_hidden, -> { where(hidden: false) }
   scope :available, -> { not_expired.not_hidden }
 
+  after_destroy :delete_courses_if_custom_courses_disabled
+  after_destroy :delete_cache
```

Comment:
maybe not allow full deletion; maybe just add deleted_at: datetime and if set, consider it to be deleted.. maybe